rust/storage/src/admissioncontrolleds3.rs

```diff
@@ -255,6 +336,9 @@ impl AdmissionControlledS3Storage {
             };
         }
 
+        let _guard =
```

Comment:
How does this work? This does not notify the task that's waiting to acquire the token, so how does it increase/decrease its priority?